x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlExpressionTranslators.java

```diff
@@ -531,6 +532,17 @@ private static RangeQuery translate(Range r, TranslatorHandler handler) {
     public static class MatchFunctionTranslator extends ExpressionTranslator<Match> {
         @Override
         protected Query asQuery(Match match, TranslatorHandler handler) {
+            // This is only temporary until the MatchQueryBuilder has support for semantic_text.
+            // At that point we can simply remove SemanticQuery and rely on MatchQuery.
+            if (match.field().dataType() == DataType.SEMANTIC_TEXT) {
+                return new SemanticQuery(
+                    match.source(),
+                    ((FieldAttribute) match.field()).name(),
+                    match.queryAsText(),
+                    match.inferenceResults()
+                );
+            }
+
```

Comment:
There is another scenario that I can think of - there are two indices with the same field name, one is semantic text, the other is a text field for example. If I query each of them individually, I got expected results, however if I query both of them together, I got empty results. It is similar to an issue that I brought up to Carlos on his PR, however, semantic text has a different type of query for now, perhaps it has to be taken care of in a different way.

```
+ curl -u elastic:password -v -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "from testidx00 | where match(textfield, \"live\")"
}
'

          textfield           
------------------------------
live long and prosper 00 elser

+ curl -u elastic:password -v -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "from testidx04  | where match(textfield, \"live\")"
}
'

              textfield              |          textfield.keyword          
-------------------------------------+-------------------------------------
live long and prosper 04 text keyword|live long and prosper 04 text keyword

+ curl -u elastic:password -v -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "from testidx00, testidx04 | where match(textfield::keyword, \"live\")  | eval k = textfield::keyword | keep k"
}
'

       k       
---------------
```

Reply:
Good find! I added this as a follow up of things to do before we take this out of snapshot in https://github.com/elastic/elasticsearch/issues/115103.